core/src/main/java/com/datastax/oss/driver/api/core/type/reflect/GenericType.java

```diff
@@ -251,6 +252,7 @@ public final boolean isPrimitive() {
    * Returns the array component type if this type represents an array ({@code int[]}, {@code T[]},
    * {@code <? extends Map<String, Integer>[]>} etc.), or else {@code null} is returned.
    */
+  @Nullable
   @SuppressWarnings("unchecked")
   public final GenericType<?> getComponentType() {
     return new GenericType(token.getComponentType());
```

Comment:
This produces a warning because `new GenericType(...)` is never null. Which is actually the symptom of a bug: `token.getComponentType()` can be null, this must be handled.
I'm fixing it directly.

Reply:
Ah right, I didn't even consider that, good catch!